integration_tests/src/main/python/noop_test.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from asserts import *
+from data_gen import *
+from marks import incompat, ignore_order
+from spark_session import *
+
+
+def test_noop_format_write():
+    """Test for noop format write support - should work on GPU after our changes"""
+    def test_noop_write(spark):
+        data = [("Alice", 34), ("Bob", 45), ("Cathy", 29)]
+        columns = ["Name", "Age"]
+        df = spark.createDataFrame(data, columns)
+        # This should work on GPU after our implementation
+        df.write.format("noop").mode("overwrite").save()
+    
+    # Test that it runs on GPU without falling back to CPU
+    with_gpu_session(test_noop_write, conf={"spark.rapids.sql.enabled": "true"})
+
+
+@pytest.mark.parametrize("mode", ["append", "overwrite", "ignore"])
```

Comment:
On the CPU if I try to write in ignore mode I get an error.

```
scala> spark.range(100).write.format("noop").mode("ignore").save()
org.apache.spark.sql.AnalysisException: TableProvider implementation noop cannot be written with Ignore mode, please use Append or Overwrite modes instead.
  at org.apache.spark.sql.errors.QueryCompilationErrors$.writeWithSaveModeUnsupportedBySourceError(QueryCompilationErrors.scala:2881)
  at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:349)
  at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:247)
  ... 47 elided
```

The others work.